### ISSUE TITLE
在使用自己的数据集训练时，在验证时的结果全为0

### ISSUE THREAD

**Issue by Dominic23331** [open-mmlab/mmpose#2019]
当我使用自己的数据集训练时，在验证轮次时，出现的结果全部为0

 Average Precision  (AP) @[ IoU=0.50:0.95 | area=   all | maxDets= 20 ] =  0.000
 Average Precision  (AP) @[ IoU=0.50      | area=   all | maxDets= 20 ] =  0.000
 Average Precision  (AP) @[ IoU=0.75      | area=   all | maxDets= 20 ] =  0.000
 Average Precision  (AP) @[ IoU=0.50:0.95 | area=medium | maxDets= 20 ] = -1.000
 Average Precision  (AP) @[ IoU=0.50:0.95 | area= large | maxDets= 20 ] = -1.000
 Average Precision  (AP) @[ IoU=0.50      | area=   all | maxDets= 20 ] =  0.000
 Average Precision  (AP) @[ IoU=0.75      | area=   all | maxDets= 20 ] =  0.000
 Average Precision  (AP) @[ IoU=0.50:0.95 | area=medium | maxDets= 20 ] = -1.000
 Average Precision  (AP) @[ IoU=0.50:0.95 | area= large | maxDets= 20 ] = -1.000
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets= 20 ] =  0.000
 Average Recall     (AR) @[ IoU=0.50      | area=   all | maxDets= 20 ] =  0.000
 Average Recall     (AR) @[ IoU=0.75      | area=   all | maxDets= 20 ] =  0.000
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=medium | maxDets= 20 ] = -1.000
 Average Recall     (AR) @[ IoU=0.50:0.95 | area= large | maxDets= 20 ] = -1.000
03/04 22:07:46 - mmengine - INFO - Epoch(val) [210][6/6]  coco/AP: 0.000000  coco/AP .5: 0.000000  coco/AP .75: 0.000000  coco/AP (M): -1.000000  coco/AP (L): -1.000000  coco/AR: 0.000000  coco/AR .5: 0.000000  coco/AR .75: 0.000000
 coco/AR (M): -1.000000  coco/AR (L): -1.000000

以下是配置文件信息
_base_ = ['../../../_base_/default_runtime.py']

# runtime
train_cfg = dict(max_epochs=210, val_interval=10)

# optimizer
optim_wrapper = dict(optimizer=dict(
    type='Adam',
    lr=5e-4,
))

# learning policy
param_scheduler = [
    dict(
        type='LinearLR', begin=0, end=500, start_factor=0.001,
        by_epoch=False),  # warm-up
    dict(
        type='MultiStepLR',
        begin=0,
        end=210,
        milestones=[170, 200],
        gamma=0.1,
        by_epoch=True)
]

# automatically scaling LR based on the actual training batch size
auto_scale_lr = dict(base_batch_size=512)

# hooks
default_hooks = dict(checkpoint=dict(save_best='coco/AP', rule='greater'))

# codec settings
codec = dict(
    type='MSRAHeatmap',
    input_size=(192, 256),
    heatmap_size=(48, 64),
    sigma=2,
    unbiased=True)

# model settings
model = dict(
    type='TopdownPoseEstimator',
    data_preprocessor=dict(
        type='PoseDataPreprocessor',
        mean=[123.675, 116.28, 103.53],
        std=[58.395, 57.12, 57.375],
        bgr_to_rgb=True),
    backbone=dict(
        type='HRNet',
        in_channels=3,
        extra=dict(
            stage1=dict(
                num_modules=1,
                num_branches=1,
                block='BOTTLENECK',
                num_blocks=(4, ),
                num_channels=(64, )),
            stage2=dict(
                num_modules=1,
                num_branches=2,
                block='BASIC',
                num_blocks=(4, 4),
                num_channels=(32, 64)),
            stage3=dict(
                num_modules=4,
                num_branches=3,
                block='BASIC',
                num_blocks=(4, 4, 4),
                num_channels=(32, 64, 128)),
            stage4=dict(
                num_modules=3,
                num_branches=4,
                block='BASIC',
                num_blocks=(4, 4, 4, 4),
                num_channels=(32, 64, 128, 256))),
        init_cfg=dict(
            type='Pretrained',
            checkpoint='https://download.openmmlab.com/mmpose/'
            'pretrain_models/hrnet_w32-36af842e.pth'),
    ),
    head=dict(
        type='HeatmapHead',
        in_channels=32,
        out_channels=10,
        deconv_out_channels=None,
        loss=dict(type='KeypointMSELoss', use_target_weight=True),
        decoder=codec),
    test_cfg=dict(
        flip_test=True,
        flip_mode='heatmap',
        shift_heatmap=True,
    ))

# base dataset settings
dataset_type = 'KidDataset'
data_mode = 'topdown'
data_root = 'data/coco/'

# pipelines
train_pipeline = [
    dict(type='LoadImage', file_client_args={{_base_.file_client_args}}),
    dict(type='GetBBoxCenterScale'),
    dict(type='RandomFlip', direction='horizontal'),
    dict(type='RandomHalfBody'),
    dict(type='RandomBBoxTransform'),
    dict(type='TopdownAffine', input_size=codec['input_size']),
    dict(type='GenerateTarget', target_type='heatmap', encoder=codec),
    dict(type='PackPoseInputs')
]
val_pipeline = [
    dict(type='LoadImage', file_client_args={{_base_.file_client_args}}),
    dict(type='GetBBoxCenterScale'),
    dict(type='TopdownAffine', input_size=codec['input_size']),
    dict(type='PackPoseInputs')
]

# data loaders
train_dataloader = dict(
    batch_size=4,
    num_workers=2,
    persistent_workers=True,
    sampler=dict(type='DefaultSampler', shuffle=True),
    dataset=dict(
        type=dataset_type,
        data_root=data_root,
        data_mode=data_mode,
        ann_file='annotations/keypoints_train.json',
        data_prefix=dict(img='train/'),
        pipeline=train_pipeline,
    ))
val_dataloader = dict(
    batch_size=32,
    num_workers=2,
    persistent_workers=True,
    drop_last=False,
    sampler=dict(type='DefaultSampler', shuffle=False, round_up=False),
    dataset=dict(
        type=dataset_type,
        data_root=data_root,
        data_mode=data_mode,
        ann_file='annotations/keypoints_val.json',
        # bbox_file='data/coco/person_detection_results/'
        # 'COCO_val2017_detections_AP_H_56_person.json',
        data_prefix=dict(img='val/'),
        test_mode=True,
        pipeline=val_pipeline,
    ))
test_dataloader = val_dataloader

# evaluators
val_evaluator = dict(
    type='CocoMetric',
    ann_file=data_root + 'annotations/keypoints_val.json')
test_evaluator = val_evaluator

**Comment by Ben-Louis** [comment]
您好，感谢您的关注和使用！直接看 config 我们也没办法直接定位问题所在，建议尝试以下方式排查问题：

1. 参考 [可视化文档](https://mmpose.readthedocs.io/en/dev-1.x/user_guides/visualization.html#browse-dataset) 来可视化您的训练集、测试集，确保数据的标注、处理符合预期；
2. 对于训练好的模型，可以参考 [人体姿态检测demo](https://github.com/open-mmlab/mmpose/blob/dev-1.x/demo/docs/2d_human_pose_demo.md#use-mmdet-for-human-bounding-box-detection) 来可视化模型的预测结果。通过这一步可能可以确定问题是出在训练还是测试阶段

**Comment by Dominic23331** [comment]
经测试，只有训练阶段验证精度时出现问题，网络训练后是有效的

**Comment by Ben-Louis** [comment]
如果网络推理结果正常的话，可以看下测试数据集有没有问题：
1. 通过上面说的方法可视化测试集数据，看标注是否正确；
2. 检查 metainfo ，包括是否在数据集类中正确设置、`sigmas`, `swap` 等字段的值是否正确；
3. 检查测试集标注文件格式和 [COCO数据集格式](https://cocodataset.org/#format-data) 是否一致

**Comment by Dominic23331** [comment]
已解决，谢谢

**Comment by xjtbinghan** [comment]
作者您好，我遇到了和您一样的问题，可以请教您是如何解决的吗，感谢！

**Comment by jackthelad0312** [comment]
> 已解决，谢谢

佬，怎么解决的呀？

**Comment by He-ningbo** [comment]
我也遇到了这个问题，请问大佬怎么解决的

**Comment by Dominic23331** [comment]
> 我也遇到了这个问题，请问大佬怎么解决的

你检查一下数据集的格式

**Comment by Jazzy0302** [comment]
大佬 怎么解决的，能不能详细讲讲，数据集格式问题具体是什么

**Comment by LinfieWang** [comment]
我也遇到了相同的问题，训练过程中损失正常下降，acc_pose逐渐上升，但在验证集上的AP和AR始终全为0。请问这是什么问题呢？

01/13 22:08:07 - mmengine - INFO - Epoch(train) [415][12/12]  base_lr: 2.054913e-04 lr: 2.054913e-04  eta: 0:04:07  time: 4.389399  data_time: 3.921792  memory: 26662  loss: 0.200440  loss_kpt: 0.200440  acc_pose: 0.994949
01/13 22:08:56 - mmengine - INFO - Exp name: rtmpose-l_8xb256-420e_coco-surgery-256x192_20250113_161444
01/13 22:08:56 - mmengine - INFO - Epoch(train) [416][12/12]  base_lr: 2.035439e-04 lr: 2.035439e-04  eta: 0:03:18  time: 4.376618  data_time: 3.912001  memory: 26662  loss: 0.200207  loss_kpt: 0.200207  acc_pose: 1.000000
01/13 22:09:26 - mmengine - INFO - Exp name: rtmpose-l_8xb256-420e_coco-surgery-256x192_20250113_161444
01/13 22:09:46 - mmengine - INFO - Exp name: rtmpose-l_8xb256-420e_coco-surgery-256x192_20250113_161444
01/13 22:09:46 - mmengine - INFO - Epoch(train) [417][12/12]  base_lr: 2.020209e-04 lr: 2.020209e-04  eta: 0:02:28  time: 4.387940  data_time: 3.926408  memory: 26662  loss: 0.199547  loss_kpt: 0.199547  acc_pose: 1.000000
01/13 22:10:35 - mmengine - INFO - Exp name: rtmpose-l_8xb256-420e_coco-surgery-256x192_20250113_161444
01/13 22:10:35 - mmengine - INFO - Epoch(train) [418][12/12]  base_lr: 2.009227e-04 lr: 2.009227e-04  eta: 0:01:39  time: 4.407654  data_time: 3.944402  memory: 26662  loss: 0.201430  loss_kpt: 0.201430  acc_pose: 1.000000
01/13 22:11:25 - mmengine - INFO - Exp name: rtmpose-l_8xb256-420e_coco-surgery-256x192_20250113_161444
01/13 22:11:25 - mmengine - INFO - Epoch(train) [419][12/12]  base_lr: 2.002495e-04 lr: 2.002495e-04  eta: 0:00:49  time: 4.370397  data_time: 3.910620  memory: 26662  loss: 0.200332  loss_kpt: 0.200332  acc_pose: 1.000000
01/13 22:12:16 - mmengine - INFO - Exp name: rtmpose-l_8xb256-420e_coco-surgery-256x192_20250113_161444
01/13 22:12:16 - mmengine - INFO - Epoch(train) [420][12/12]  base_lr: 2.000015e-04 lr: 2.000015e-04  eta: 0:00:00  time: 4.360272  data_time: 3.898984  memory: 26662  loss: 0.199742  loss_kpt: 0.199742  acc_pose: 1.000000
01/13 22:12:16 - mmengine - INFO - Saving checkpoint at 420 epochs
01/13 22:12:31 - mmengine - INFO - Evaluating CocoMetric...
Loading and preparing results...
DONE (t=0.03s)
creating index...
index created!
Running per image evaluation...
Evaluate annotation type *keypoints*
DONE (t=0.51s).
Accumulating evaluation results...
DONE (t=0.02s).
 Average Precision  (AP) @[ IoU=0.50:0.95 | area=   all | maxDets= 20 ] =  0.000
 Average Precision  (AP) @[ IoU=0.50      | area=   all | maxDets= 20 ] =  0.000
 Average Precision  (AP) @[ IoU=0.75      | area=   all | maxDets= 20 ] =  0.000
 Average Precision  (AP) @[ IoU=0.50:0.95 | area=medium | maxDets= 20 ] = -1.000
 Average Precision  (AP) @[ IoU=0.50:0.95 | area= large | maxDets= 20 ] = -1.000
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets= 20 ] =  0.000
 Average Recall     (AR) @[ IoU=0.50      | area=   all | maxDets= 20 ] =  0.000
 Average Recall     (AR) @[ IoU=0.75      | area=   all | maxDets= 20 ] =  0.000
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=medium | maxDets= 20 ] = -1.000
 Average Recall     (AR) @[ IoU=0.50:0.95 | area= large | maxDets= 20 ] = -1.000
01/13 22:12:32 - mmengine - INFO - Epoch(val) [420][30/30]    coco/AP: 0.000000  coco/AP .5: 0.000000  coco/AP .75: 0.000000  coco/AP (M): -1.000000  coco/AP (L): -1.000000  coco/AR: 0.000000  coco/AR .5: 0.000000  coco/AR .75: 0.000000  coco/AR (M): -1.000000  coco/AR (L): -1.000000  data_time: 0.294721  time: 0.392304
